test/unit/gateways/payflow_test.rb

```diff
@@ -49,6 +49,70 @@ def test_successful_authorization
     refute response.fraud_review?
   end
 
+  def test_successful_purchase_with_stored_credential
+    @options[:stored_credential] = {
+      initial_transaction: true,
+      reason_type: 'recurring',
+      initiator: 'cardholder',
+      network_transaction_id: nil
+    }
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |_endpoint, data, _headers|
+      assert_match %r(<CardOnFile>CITR</CardOnFile>), data
+    end.respond_with(successful_purchase_with_fraud_review_response)
+
+    @options[:stored_credential] = {
+      initial_transaction: true,
```

Comment:
Again, I don't think we can have `initial_transaction = true` and `initiator = 'merchant'`.

Reply:
Updated 👍